website/src/parsers/js/recast.js

```diff
@@ -68,7 +68,9 @@ export default {
       case 'babel5':
         options.parser = parsers[options.parser];
         break;
-      case 'esprima':
+      case 'typescript':
+        options.parser = require('recast/parsers/typescript')
```

Comment:
This needs to be moved into the `loadParser` method, otherwise the module is going to be bundled with the core instead of being lazy loaded when recast is selected.